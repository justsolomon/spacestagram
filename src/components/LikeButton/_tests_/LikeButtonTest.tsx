import { useState } from "react";
import LikeButton from "..";

const LikeButtonTest = () => {
  const [isLiked, setIsLiked] = useState<boolean>(false);

  return (
    <LikeButton
      liked={isLiked}
      title="test"
      toggleLike={() => setIsLiked(!isLiked)}
    />
  );
};

export default LikeButtonTest;
