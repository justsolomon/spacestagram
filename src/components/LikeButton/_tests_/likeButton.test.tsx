import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import LikeButtonTest from "./LikeButtonTest";

describe("<LikeButton />", () => {
  it("should render the like button properly", () => {
    const { asFragment } = render(<LikeButtonTest />);

    expect(asFragment).toMatchSnapshot();
  });
  it("should like and unlike an item", () => {
    render(<LikeButtonTest />);

    const likeButton = screen.getByTestId("like-button");
    expect(likeButton).toHaveAccessibleName("Like test photo");
    expect(likeButton).not.toHaveClass("like-button--active");

    userEvent.click(likeButton);
    expect(likeButton).toHaveAccessibleName("Unlike test photo");
    expect(likeButton).toHaveClass("like-button--active");

    userEvent.click(likeButton);
    expect(likeButton).toHaveAccessibleName("Like test photo");
    expect(likeButton).not.toHaveClass("like-button--active");
  });
});
