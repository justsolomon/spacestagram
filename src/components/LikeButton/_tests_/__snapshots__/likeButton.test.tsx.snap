// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LikeButton /> should render the like button properly 1`] = `[Function]`;
